### PR TITLE
Drag and drop files into the editor

### LINKED ISSUE #1227: Drag and Drop files to Editor
So what if you're on the image or sound editor and you wanted to drag and drop the files to the editor? This should be part of the developer tools.

### PR DESCRIPTION
**Pull request by SheepTester**
**Resolves**

Resolves #1227 

**Changes**

I added the ability to drag files into the costume/sound list, the sprites pane, and the stage sidebar thing. This inserts the dropped files into the same file input that is used by the "Upload Xyz" option in the slidey "Choose a Xyz" menu.

Dragging files into the costume and sound list

Dragging files into the sprites list and stage

**Reason for changes**

Some people, especially those not super versed with computers, prefer dragging files in from their file explorer instead of having to hunt through folders in a file selection window.

**Tests**

I dragged in various files into the drop areas I listed above. If you drag in a file that isn't supported, nothing happens. I'm not sure whether this will be incompatible with other addons, though.

### CODE REVIEW

**Review by Explosion-Scratch — APPROVED** [review]
Wow! I'm impressed at how efficient your code is! =D This addon looks great!

File: addons/drag-drop/userscript.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+export default async function ({ addon, global, console }) {
+  const DRAG_AREA_CLASS = "sa-drag-area";
+  const DRAG_OVER_CLASS = "sa-dragged-over";
```
Comment (RedGuy12) [inline addons/drag-drop/userscript.js:3]: i think we're trying not to make our own classes

File: addons/drag-drop/userscript.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  }
+
+  droppable(
+    addon.tab.waitForElement('div[class*="sprite-selector_sprite-selector"]'),
+    addon.tab.waitForElement('div[class*="sprite-selector_sprite-selector"] input[class*="action-menu_file-input"]')
+  );
+  droppable(
+    addon.tab.waitForElement('div[class*="stage-selector_stage-selector"]'),
+    addon.tab.waitForElement('div[class*="stage-selector_stage-selector"] input[class*="action-menu_file-input"]')
+  );
```
Comment (TheColaber) [inline addons/drag-drop/userscript.js:32]: does this still work when exitting and reentering the editor?

**Review by RedGuy12 — CHANGES_REQUESTED** [review]

File: addons/drag-drop/userscript.js (hunk 1 above)
Comment (TheColaber) [reply addons/drag-drop/userscript.js:3]: See how it is used in the css... this is perfectly fine.

File: addons/drag-drop/addon.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    }
+  ],
+  "userstyles": [
+    {
+      "url": "dragged-over.css",
+      "matches": ["https://scratch.mit.edu/projects/*"]
+    }
+  ],
+  "tags": ["editor"],
+  "enabledByDefault": false
```
Comment (WorldLanguages) [inline addons/drag-drop/addon.json:22]: Add "l10n": true

File: addons/drag-drop/userscript.js (hunk 2 above)
Comment (TheColaber) [reply addons/drag-drop/userscript.js:32]: This does not work when exitting and reentering the editor. Please fix this.

**Review by retronbv — APPROVED** [review]
It’s good, just add `"l10n": true`

**Comment by WorldLanguages** [comment]
Excellent, not sure how no one had done this until now - code is so simple and clever as well.

**Pull request merged**